proj_utils/Validator.py

```diff
@@ -0,0 +1,201 @@
+              "type": "object",
+              "properties": {
+                "subject": {
+                  "type": "string"
+                },
+                "grades": {
+                  "type": "array",
+                  "items": [
+                    {
+                      "type": "integer"
+                    },
+                    {
+                      "type": "integer"
+                    },
+                    {
+                      "type": "integer"
+                    }
+                  ]
+                }
+              },
+              "required": [
+                "subject",
+                "grades"
+              ]
+            }
+          ]
+        },
+        "birthDate": {
+          "type": "string"
+        },
+        "age": {
+          "type": "integer"
+        },
+        "gender": {
+          "type": "string"
+        },
+        "behaviorGrade": {
+          "type": "integer"
+        },
+        "notes": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "studentDetails",
+        "subjectGrades",
+        "birthDate",
+        "age",
+        "gender",
+        "behaviorGrade",
+        "notes"
+      ]
+    }
+
+    def __init__(self, body):
+        self.json_file = json.loads(body)
+        self.message = ""
+        self.is_valid_schema = Draft3Validator(Validator.schema).is_valid(self.json_file)
+
+    def is_valid_firstname(self, firstName):
+        if not firstName.isalpha():
+            self.message += "Invalid First Name !\n"
+            return False
+        return True
+
+    def is_valid_lastname(self, lastName):
+        if any(char.isdigit() for char in lastName):
+            self.message += "Invalid Last Name !\n"
+            return False
+        return True
+
+    def is_valid_id(self, id):
+        if len(str(id)) != 9 or not str(id).isnumeric():
+            self.message += "Invalid id !\n"
+            return False
+        return True
+
+    def is_valid_birthdate(self, birthDate):
+        try:
+            datetime.datetime.strptime(str(birthDate), '%d/%m/%Y')
+        except ValueError:
+            self.message += "Invalid birth date !\n"
+            return False
+        return True
+
+    def is_valid_age(self, age, birthDate):
+        if (datetime.datetime.now() - datetime.datetime.strptime(birthDate, "%d/%m/%Y")).days // 365 != age:
+            self.message += "Invalid age !\n"
+            return False
+        return True
+
+    def is_valid_gender(self, gender):
+        if gender not in ["נקבה", "אחר", "זכר"]:
+            self.message += "Invalid gender !\n"
+            return False
+        return True
+
+    def is_valid_behaviour_grade(self, behaviourGrade):
+        if behaviourGrade not in [i for i in range(1, 11)]:
+            self.message += "Invalid behaviour grade !\n"
+            return False
+        return True
+
+    def are_grades_valid(self, subjects):
+        for subject in subjects:
```

Comment:
you are coupling your validation to the current format of the json. you shouldn't do that